source/extensions/filters/common/ext_authz/ext_authz_http_impl.cc

```diff
@@ -50,13 +51,19 @@ struct SuccessResponse {
                 Http::LowerCaseString{std::string(header.key().getStringView())},
                 std::string(header.value().getStringView()));
           }
+          if (context->append_matchers_->matches(header.key().getStringView())) {
+            context->response_->headers_to_add_and_append.emplace_back(
```

Comment:
Add comment telling that we're appending (if there is an existing key). To explain the behavior of it.